views/Products/Create/Form.tsx

```diff
@@ -59,21 +66,27 @@ const CreateProductForm: FC<MutableProductFormProps> = ({ onSubmit, loading }) =
       </label>
       <div className="container-price">
         <p className="price">S/.</p>
-        <input
-          className="input-product font-visby"
+        <CurrencyFormat
+          fixedDecimalScale={true}
+          decimalScale={2}
+          className={`input-product font-visby ${errors.basePriceSol && 'error'}`}
           type="text"
+          allowNegative={false}
+          isNumericString={true}
           id="basePriceSol"
           placeholder="00.00"
-          {...register('basePriceSol', { required: { value: true, message: 'Debe colocar un precio' } })}
+          onValueChange={handlePrice}
+          {...register('basePriceSol', {
+            required: true,
+            validate: value => value !== 0,
+          })}
```

Comment:
If is already registered really is needed the handlePrice function?, it's not updating the value automatically?

Reply:
I think because the value of the CurrencyFormat component is an object, I could not pass an object directly through register. So with setValue, it saves better